Source/ASCollectionNode.mm

```diff
@@ -30,17 +30,20 @@
 
 #pragma mark - _ASCollectionPendingState
 
-@interface _ASCollectionPendingState : NSObject
-@property (weak, nonatomic) id <ASCollectionDelegate>   delegate;
-@property (weak, nonatomic) id <ASCollectionDataSource> dataSource;
+@interface _ASCollectionPendingState : NSObject {
+@public
+  std::vector<std::vector<ASRangeTuningParameters>> _tuningParameters;
```

Comment:
Not specific to this diff, but we should probably use `ASRangeTuningParameters[ASLayoutRangeModeCount][ASLayoutRangeTypeCount] _tuningParameters` here so we get static allocation.